packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/ParameterProvider.cs

```diff
@@ -175,5 +204,52 @@ private ParameterValidationType GetParameterValidation()
 
             return ParameterValidationType.AssertNotNull;
         }
+
+        private static ValueExpression? GetParameterDefaultValue(InputParameter inputParameter)
+        {
+            if (inputParameter.DefaultValue is null)
+            {
+                return null;
+            }
+
+            var defaultValue = inputParameter.DefaultValue.Value;
+            CSharpType valueType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(inputParameter.DefaultValue.Type);
+
+            // handle default values for collections
+            if (defaultValue == null && valueType.IsCollection)
+            {
+                return New.Instance(valueType);
+            }
+            else if (valueType.IsFrameworkType && defaultValue is IConvertible)
+            {
+                // handle default values for framework types
+                var normalizedValue = Convert.ChangeType(defaultValue, valueType.FrameworkType);
+                return GetLiteralValue(normalizedValue);
+            }
+
+            return null;
+        }
+
+        private static ValueExpression? GetLiteralValue(object? value)
+        {
+            if (value == null)
+            {
+                return null;
+            }
+
+            return value switch
+            {
+                string s => Literal(s),
+                int i => Int(i),
+                long l => Long(l),
+                decimal d => Literal(d),
+                double d => Double(d),
+                float f => Float(f),
+                char c => Literal(c),
+                bool b => Bool(b),
+                BinaryData bd => Literal(bd),
```

Comment:
So for those that are not `Literal`, if you check their implementation, they are calling `Literal` and then `As<T>`, therefore we do not need to have this method, we could just call `Literal(value)`.